gb-roms/src/header/mod.rs

```diff
@@ -0,0 +1,263 @@
+				rom_size: RomSize::MByte2,
+				ram_size: RamSize::KByte8,
+				destination_code: DestinationCode::Other,
+				old_license_code: OldLicenseCode::UseNewLicenseCode,
+				rom_version: 0,
+				header_checksum: 226,
+				global_checksum: 39392
+			})
+		);
+	}
+}
+
+impl TryFrom<RawHeader> for Header {
+	type Error = Error;
+
+	fn try_from(raw: RawHeader) -> Result<Self, Self::Error> {
+		Ok(Self {
+			entry_point: raw.entry_point,
+			nitendo_logo: raw.nitendo_logo,
+			title: raw.title.try_into()?,
+			new_license_code: NewLicenseCode::try_from(
+				String::from_utf8(raw.new_license_code.into())?.as_str(),
+			)?,
+			sgb_flag: raw.sgb_flag.try_into()?,
+			cartridge_type: raw.cartridge_type.try_into()?,
+			rom_size: raw.rom_size.try_into()?,
+			ram_size: raw.ram_size.try_into()?,
+			destination_code: raw.destination_code.try_into()?,
+			old_license_code: raw.old_license_code.try_into()?,
+			rom_version: raw.rom_version,
+			header_checksum: raw.header_checksum,
+			global_checksum: u16::from_le_bytes(raw.global_checksum),
+		})
+	}
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Title {
+	Simple(String),
+	Advanced {
+		title: String,
+		manufacturer: String,
+		cgb_flag: CgbFlag,
+	},
+}
+
+impl TryFrom<[u8; 16]> for Title {
+	type Error = Error;
+
+	fn try_from(raw: [u8; 16]) -> Result<Self, Self::Error> {
+		if raw[15] == 0 {
+			Ok(Title::Simple(
+				String::from_utf8(raw.into())?
+					.trim_end_matches(char::from(0))
+					.to_string(),
+			))
+		} else {
+			Ok(Title::Advanced {
+				title: String::from_utf8(raw[0..10].into())?
+					.trim_end_matches(char::from(0))
+					.to_string(),
+				manufacturer: String::from_utf8(raw[10..14].into())?
+					.trim_matches(char::from(0))
+					.to_string(),
+				cgb_flag: raw[15].try_into()?,
+			})
+		}
+	}
+}
+
+/// The cartridge Header is at 0x100-0x14f
+#[derive(Debug)]
+#[repr(C)]
+pub struct RawHeader {
+	pub entry_point: [u8; 4],
+	pub nitendo_logo: [u8; 48],
+	pub title: [u8; 16],
+	pub new_license_code: [u8; 2],
+	pub sgb_flag: u8,
+	pub cartridge_type: u8,
+	pub rom_size: u8,
+	pub ram_size: u8,
+	pub destination_code: u8,
+	pub old_license_code: u8,
+	pub rom_version: u8,
+	pub header_checksum: u8,
+	pub global_checksum: [u8; 2],
+}
+
+impl From<&[u8; 80]> for RawHeader {
+	fn from(chunk: &[u8; 80]) -> Self {
+		Self {
+			entry_point: <[u8; 4]>::try_from(&chunk[..4]).unwrap(),
+			nitendo_logo: <[u8; 48]>::try_from(&chunk[4..52]).unwrap(),
+			title: <[u8; 16]>::try_from(&chunk[52..68]).unwrap(),
+			new_license_code: <[u8; 2]>::try_from(&chunk[68..70]).unwrap(),
+			sgb_flag: chunk[70],
+			cartridge_type: chunk[71],
+			rom_size: chunk[72],
+			ram_size: chunk[73],
+			destination_code: chunk[74],
+			old_license_code: chunk[75],
+			rom_version: chunk[76],
+			header_checksum: chunk[77],
+			global_checksum: <[u8; 2]>::try_from(&chunk[78..80]).unwrap(),
+		}
+	}
+}
+
+#[test]
+fn test_raw_header_size() {
+	assert_eq!(std::mem::size_of::<RawHeader>(), 80)
+}
```

Comment:
For the root file of folder modules we have the choice between `src/module/mod.rs` and `src/module.rs`.  
I think the second option make it easier to understand what the file does.  
We should decide collectively about which one we use.

Reply:
Usage of mod.rs is discouraged since it quickly makes everything very confusing: 
https://doc.rust-lang.org/reference/items/modules.html#module-source-filenames

Reply:
I've renamed to `header.rs`